backend/src/question/question.service.ts

```diff
@@ -0,0 +1,62 @@
+import { Injectable } from "@nestjs/common";
+import { Category, Question, Topic } from "@prisma/client";
+
+import { PrismaService } from "src/core/prisma.service";
+
+import { DifficultyLevel } from "~shared/types/base";
+
+@Injectable()
+export class QuestionService {
+  constructor(private prisma: PrismaService) {}
+
+  async getQuestionById(id: number): Promise<Question | null> {
+    return this.prisma.question.findUnique({
+      where: { id: id },
+    });
+  }
+
+  async getTopicsByQuestionId(id: number): Promise<Topic[] | null> {
+    return this.prisma.topic.findMany({
+      include: {
+        questions: {
+          where: {
+            id: id,
+          },
+        },
+      },
```

Comment:
Should be a where clause:

```suggestion
      where: {
        questions: {
          some: {
            id,
          },
        },
      },
```